Minicloner.Tests build fails because "target": "project" is missing from dependencies in Minicloner and Minicloner.OtherAssembly and missing global.json

## Steps to reproduce
- Run the following command:
  
  `build.ps1`
## Expected behavior

Build succeds
## Actual behavior

Build fails at the **Restore** step with the following message:

```
2016-07-31T16:20:32.3075999Z ##[error]Errors in C:\a\1\s\Minicloner\test\Minicloner.Tests\project.json
2016-07-31T16:20:32.3075999Z ##[error]    Unable to resolve 'Minicloner (>= 1.0.0)' for '.NETCoreApp,Version=v1.0'.
2016-07-31T16:20:32.3486020Z ##[error]An error occured when executing task 'Restore'.
2016-07-31T16:20:32.3496022Z ##[error]Error: .NET Core CLI: Process returned an error (exit code 1).
2016-07-31T16:20:32.4835254Z ##[error]System.Exception: Unexpected exit code 1 returned from tool Cake.exe
2016-07-31T16:20:32.4835254Z ##[error]   at Microsoft.TeamFoundation.DistributedTask.Task.Internal.PowerShell.InvokeToolCmdlet.ProcessRecord()
2016-07-31T16:20:32.4845260Z ##[error]   at System.Management.Automation.CommandProcessor.ProcessRecord()
```
## Environment data

```
.NET Command Line Tools (1.0.0-preview2-003121)

Product Information:
 Version:            1.0.0-preview2-003121
 Commit SHA-1 hash:  1e9d529bc5

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
```
## Solution

`/Miniclones/test/Minicloner.Tests/project.json` looks like

```
...
  "dependencies": {
    "Minicloner": "1.0.0-*",
    "Minicloner.Tests.OtherAssembly": "1.0.0-*",
    "xunit": "2.2.0-beta2-build3300",
    "dotnet-test-xunit": "2.2.0-preview2-build1029"
  },
...
```

`Minicloner` and `Minicloner.Tests.OtherAssembly` are not being specified as **project** but as the default **package** so it tries to go to a package feed to restore it.

It should look like:

```
...
  "dependencies": {
    "Minicloner": {
      "version": "1.0.0-*",
      "target": "project"
    },
    "Minicloner.Tests.OtherAssembly": {
      "version": "1.0.0-*",
      "target": "project"
    },
    "xunit": "2.2.0-beta2-build3300",
    "dotnet-test-xunit": "2.2.0-preview2-build1029"
  },
...
```

Also for `dotnet build` to find the source of `Minicloner` project, a `global.json` with the following content must be created:

```
{
  "projects": [ "src", "test" ]
}
```

More info: https://docs.microsoft.com/en-us/dotnet/articles/core/tutorials/libraries#how-to-use-multiple-projects

--- a/Minicloner/Minicloner.sln
+++ b/Minicloner/Minicloner.sln
@@ -8,6 +8,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution 
 		.gitignore = .gitignore
 		build.cake = build.cake
 		build.ps1 = build.ps1
+		global.json = global.json
 	EndProjectSection
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{A622D4FB-F3A9-406A-A800-0B0FF68D92A5}"

--- a/Minicloner/global.json
+++ b/Minicloner/global.json
@@ -1,0 +1,3 @@
+{
+  "projects": [ "src", "test" ]
+}

--- a/Minicloner/test/Minicloner.Tests/project.json
+++ b/Minicloner/test/Minicloner.Tests/project.json
@@ -3,8 +3,14 @@
   "testRunner": "xunit",
 
   "dependencies": {
-    "Minicloner": "1.0.0-*",
-    "Minicloner.Tests.OtherAssembly": "1.0.0-*",
+    "Minicloner": {
+      "version": "1.0.0-*",
+      "target": "project"
+    },
+    "Minicloner.Tests.OtherAssembly": {
+      "version": "1.0.0-*",
+      "target": "project"
+    },
     "xunit": "2.2.0-beta2-build3300",
     "dotnet-test-xunit": "2.2.0-preview2-build1029"
   },

